[Chew Jun Wei Max] iP

MuxPotato opened the pull request.

MuxPotato committed:
Add Increment: Level-0 Greet

MuxPotato committed:
Add user command and respond to various user command

MuxPotato committed:
corrected echoing commands

MuxPotato committed:
add list feature

MuxPotato committed:
add task, add unmark/mark feature

MuxPotato committed:
Add Deadline, Event and Todo as subclass of Task

KN-CY commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+                break;
+
+
+            case "deadline":
+                System.out.println("Got it. I've added this task:"); // shift this line below with the another print statement later
+                taskString = GetTaskString(userInput,userCommand);
+                taskName = GetDeadlineTaskName(taskString);
+
+                String deadlineDueDate = GetDeadlineDueDateString(taskString);
+                Deadline newDeadlineTask = new Deadline(taskName,deadlineDueDate);
+                userTasks = AddUserTask(userTasks, newDeadlineTask);
+                System.out.println(userTasks[userTasks.length-1]);
+                System.out.println("Now you have " + userTasks.length + " in the list.");
+                break;
```
Could consider extracting each of these case for more abstraction

briantjs00 replied:
I agree that maybe you could consider constructing methods for each cases

KN-CY commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+                break;
+            }
+        }
+
+
+
+
```
Can consider removing extra blank lines.

briantjs00 replied:
I agree, can consider standardising the empty line spacing between each cases and the extra empty lines to make the code look neater.

KN-CY commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+
+    }
+
+    private static Task[] AddUserTask(Task[] userTasks, Task newTask) {
```
Method names should be in camelCase.

briantjs00 replied:
I agree with KN-CY, methods should be written in camelCase. But good job with the naming of methods

KN-CY commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+        return taskName;
+    }
+
+    private static String GetEventFromDate (String taskString) {
+        int slashIndex = taskString.indexOf("/");
+        int secondSlashIndex = taskString.indexOf("/",slashIndex+1);
+        String fromDate = taskString.substring(slashIndex+5,secondSlashIndex); // + 4 to account for the "from"
+        fromDate = fromDate.trim();
+        return fromDate;
+    }
```
Good job providing abstraction for the various types of string parsing.

KN-CY reviewed (commented):
Overall code is readable, of good quality, and follows most standards. But could possibly be improved as commented.

briantjs00 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,40 @@
+    public void setisDone(Boolean isDone) {
+        this.isDone = isDone;
+    }
+
```
Maybe should insert "@Override" here for the toString method. Do it for the other subclasses of Task as well.

briantjs00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,179 @@
+import java.util.Arrays;
```
Good job in listing the imported classes explicitly

briantjs00 reviewed (commented):
Overall, the code follows most of the coding standards, and the extraction of string literals to static makes the code easy to read.
Well done!

MuxPotato committed:
Extracted case of user command inputs from main into seperate functions
Ensure one space in between methods
Added @Override symbol for Todo, Deadline and Event file to indicate override of Todo's toString() function
Ensured methods names are camelcase
Changed processing of tasks strings

MuxPotato committed:
Fixed spacing in Duke

MuxPotato committed:
Added Exceptions for invalid user input

MuxPotato committed:
Merge branch 'branch-Level-5'

qianz-z commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,223 @@
+
+        String userInput;
+        Scanner in = new Scanner(System.in);
+        Task[] userTasks = new Task[0];
```
Try to avoid magic literals. There has been a few more instances in the rest of your code. You can use a named constant makes the code easier to understand because the name tells us more about the meaning of the number.

qianz-z commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,223 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    public static void main(String[] args)  {
```
I think you may have accidentally added an extra spacing before `{`. Do take note that it counts towards coding standards!

qianz-z commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,223 @@
+        String taskName = getEventTaskName(taskString);
+        String eventFromDate = getEventFromDate(taskString);
+        String eventToDate = getEventToDate(taskString);
+        Event newEventTask = new Event(taskName,eventFromDate,eventToDate);
```
Within your instantiation of Event, remember to add a spacing in between each inputs. And do the same for the rest of your code.

qianz-z commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,223 @@
+
+    private static void printAddedNewTask(Task[] userTasks) {
+        System.out.println("Got it. I've added this task:"); // shift this line below with the another print statement later
+        System.out.println(userTasks[userTasks.length-1]);
```
Remember to add spacing before and after `-`!

qianz-z commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,223 @@
+    private static void userCommandList(Task[] userTasks) {
+        for(int i = 0; i < userTasks.length; i++) {
+            if (userTasks[i].getisDone()) {
+                System.out.println( (i+1) + ". " + userTasks[i]);
```
In this case, in front of `(i+1)`, you have added an additional spacing in front of it. Do take note of your spaces as it counts towards coding standards!

qianz-z commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,223 @@
+    }
+    private static String getEventFromDate(String taskString) throws DukeException {
+        String[] taskStringParts = taskString.split("/from");
+        String[] taskStringParts2 = taskStringParts[1].split("/to");
```
Try to come up with a better naming for `taskStringParts2`

qianz-z reviewed (commented):
Overall, your code looks good with a few minor coding standard violations. Do take note of your spaces when coding! Good effort!

MuxPotato committed:
altered duke to better folllow java coding standards:
managed spacing between {}
spaced out arguements for functions with multiple arguements inputs
remove extra spacing for printout arguments
renamed intermediate variables for better readability

MuxPotato committed:
changed task[] array to ArrayList<task>
added delete function

MuxPotato committed:
added save function
accounted for missing file and missing directory

MuxPotato committed:
changed task array into arraylist
changed magic literals

MuxPotato committed:
Merge branch 'branch-Level-6'

MuxPotato committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java

MuxPotato committed:
added jar file

MuxPotato committed:
Fixed bug that the first task cannot be deleted

MuxPotato committed:
Refactored UI, Storage, Tasklist and Parser class

MuxPotato committed:
Added find functionality

MuxPotato committed:
Added JavaDocs

MuxPotato committed:
Updated some JavaDocs for functions

MuxPotato committed:
Merge pull request #1 from MuxPotato/branch-Level-9

Added find functionality

MuxPotato committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Ui.java

MuxPotato committed:
Merge pull request #2 from MuxPotato/branch-A-JavaDoc

Branch a java doc

MuxPotato committed:
Fixed some minor bug with event task

MuxPotato committed:
Fixed Bug for find functionality where upper cased find keywords would fail

MuxPotato committed:
First Version of User Guide

MuxPotato committed:
Updated some JavaDocs

MuxPotato committed:
Added User Guide

MuxPotato committed:
Fixed bug where no tasklist is loaded into the usertasks when save file is unable to be found or there is no savefile to begin with

MuxPotato committed:
Altered README

MuxPotato committed:
Fixed User Guide some small format

MuxPotato committed:
Fixed some problems with user guide

MuxPotato committed:
Fixed file bug
